### PR TITLE
Bookstore Set up

### PR DESCRIPTION
**Pull request by TEMESGEN-G25**
## Summary: 

- Initialised React app.
- Used to react-router-dom to link two components.
- Used props in order for child components to receive information.
- No styles have been added.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
```
Comment (brytebee) [inline public/index.html]: - [x] **Required:** It is okay to leave some phrase and/or a sentence to remember what was done with a block of code. However, it is not professional to leave multiple lines of note in your codebase. Kindly re-word them to be fewer.

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  };
+
+  return (
+    <>
+      <ul>
+        <li>
+          {title}
+          :
+          {author}
+        </li>
```
Comment (brytebee) [inline src/components/Book.js]: - [x] **Required:** You are required to render a list of books.

image

However, this is what renders to your page. Add some books.
image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @NatiG25👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- The project is initialized using create-react-app.✔️
- There are components and there are working.✔️
- Has 2 `Link`s in the navbar which leads to the booklist and categories.✔️
- The navbar might be in its own component.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 2 above)
Comment (TEMESGEN-G25) [reply src/components/Book.js]: @brytebee Thank you for your suggestions & for encouraging me to try again.
 Is it ok if I used a hook to store a list of random books in an object and render that object?

Comment (brytebee) [reply src/components/Book.js]: @NatiG25 You could as long as books display on your page on every render.

Comment (NatiG25) [reply src/components/Book.js]: @brytebee I'm sorry for the late reply but I think you got the wrong requirements. Forgive me for asking, but are you sure that these are the requirements for Module 3, Week 2, and Day 4 project?

I've done the implementations if you can please check it, thank you.

Comment (NatiG25) [reply src/components/Book.js]: @brytebee Hello again, I've checked with the requirements & it says to only render one book. So, I did the implementation but only rendered one book. 

Forgive me for asking, but can you please make sure if those are the actual requirements for Module 3, Week 2, and Day 4 project?

File: src/components/Books.js [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+      author: 'James Clear',
+    };
+  }
+
+  render() {
+    const { title, author } = this.state;
+    return (
+      <>
+        <main>
+          <Book title={title} author={author} />
```
Comment (medaminedev66) [inline src/components/Books.js]: - [x] Following the project requirements, the Books component should render a list of books. I would recommend defining an array of books and iterating over the array and passing the info of each book to the Book component. you may use the map() method for that.

Example of a books array:

```js
constructor() {
    super();
    this.books = [{title: 'Atomic Habits', author: 'James Clear',}, {title: 'Programming', author: 'Amine',}];
  }
```

Example of looping over an array of books:

```js
render() {
    return (
      <>
          {this.books.map((book)=> { 
             <Book title={book.title} author={book.author} />
          })
          }
      </>
    );
  }

File: src/components/Books.js [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+          <button type="button">Remove</button>
```
Comment (medaminedev66) [inline src/components/Books.js:37]: - [x] Following the project requirements, the books page should have a form for adding a book (no functionality yet).
Please try to render the Form component in the Books component.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @NatiG25 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js (hunk 4 above)
Comment (NatiG25) [reply src/components/Books.js:19]: @medaminedev66 Thank you for your suggestions. Just to make sure, so, I simply make the Form component a child of Books component, correct?

Comment (TEMESGEN-G25) [reply src/components/Books.js:37]: @medaminedev66 Thank you for your thorough feedback. It helped me immensely :+1: :hugs:.

**Review by roshan-bajgain — APPROVED** [review]
Hi @NatiG25, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

- [ ] Kindly add a remove buttons on a side of the title and author as shown in figure
### Figure
s


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TEMESGEN-G25** [comment]
@brytebee @medaminedev66 @roshan-bajgain Thank you for all of your reviews :+1:.



**Note**: All optional & required changes have been implemented.

**Pull request merged**